### DISCUSSION TITLE
Auto linking screenshot file to a key without OCR

### DISCUSSION THREAD

**Discussion by anudeepg** [WeblateOrg/weblate#10158] [General]
Hello,

Is there any way to automatically link a screenshot to a key (without OCR), currently I see only manual way by searching for source string and assigning a screenshot. 

Example, if we have screenshots in some folder structure that can match a component and key like below :  
Screenshots : screenshots/component/label_hello.PNG
Labels : component/labels_de.properties 
Content: 
label.hello = "Hello"

Here key : "label.hello" can automatically match to find a PNG with same key name "label.hello.PNG" or "label_hello.PNG" (dots replacing underscore ) 

Can you please guide if there is any hook/script that can do above operation or is there any option already like that ?

**Comment by nijel** [comment]
There is currently not automated way to assign screenshots to strings besides doing that using API.

**Reply by KwadroNaut** [reply]
Thanks, I can now dive into this: https://docs.weblate.org/en/latest/api.html#screenshots

**Reply by anudeepg** [reply]
Thanks for your reply, Is it possible to automate it by any feature toggle "Auto-link-screenshot-by-key-name" by any periodic task/ startup ? 
This helps everyone I guess to avoid manual operations when we have many labels with screenshots
